### PR TITLE
GH2838: MakeRelative alias

### LINKED ISSUE #2838: Add MakeRelative alias to DirectoryPath and FilePath
### What You Are Seeing?

`MakeAbsolute` is a useful alias to transform a relative path into an absolute path based on the current `WorkingDirectory`, but there's not an equivalent alias for doing the opposite, that is, transform an absolute path into a relative path.

### What is Expected?

I'd like Cake to provide a `MakeRelative` alias that transforms an absolute path into a relative path, which is useful when printing messages to the console log, for ease of reading.

e.g.
```csharp
// Environment.WorkingDirectory => "C:/Working/"

var relativePath = MakeRelative("C:/Working/build/MyApp.csproj");
// relativePath should be "./build/MyApp.csproj"
```

### What version of Cake are you using?

0.38.4

---

ps: I'm happy to send a PR that adds this feature.

### PR DESCRIPTION
**Pull request by daveMueller**
closes #2838

### CODE REVIEW

File: src/Cake.Common.Tests/Unit/IO/DirectoryAliasesTests.cs [hunk 1]
```diff
@@ -1016,6 +1017,125 @@ public void Should_Return_Absolute_Directory_Path()
...
+                AssertEx.IsArgumentNullException(result, "context");
+            }
+
+            [Fact]
+            public void Should_Throw_If_DirectoryPath_Is_Null()
+            {
+                // Given
+                var context = Substitute.For<ICakeContext>();
+                DirectoryPath path = null;
+                // When
```
Comment (patriksvensson) [inline src/Cake.Common.Tests/Unit/IO/DirectoryAliasesTests.cs:1048]: Line break

File: src/Cake.Common.Tests/Unit/IO/DirectoryAliasesTests.cs [hunk 2]
```diff
@@ -1016,6 +1017,125 @@ public void Should_Return_Absolute_Directory_Path()
...
+                AssertEx.IsArgumentNullException(result, "path");
+            }
+
+            [Fact]
+            public void Should_Throw_If_FilePath_Is_Null()
+            {
+                // Given
+                var context = Substitute.For<ICakeContext>();
+                FilePath path = null;
+                // When
```
Comment (patriksvensson) [inline src/Cake.Common.Tests/Unit/IO/DirectoryAliasesTests.cs:1061]: Line break

**Review by patriksvensson — CHANGES_REQUESTED** [review]
LGTM! Just two minor things that need to be fixed

File: src/Cake.Common.Tests/Unit/IO/DirectoryAliasesTests.cs (hunk 1 above)
Comment (daveMueller) [reply src/Cake.Common.Tests/Unit/IO/DirectoryAliasesTests.cs:1049]: Done

File: src/Cake.Common.Tests/Unit/IO/DirectoryAliasesTests.cs (hunk 2 above)
Comment (daveMueller) [reply src/Cake.Common.Tests/Unit/IO/DirectoryAliasesTests.cs:1063]: Done

**Review by devlead — APPROVED** [review]
LGTM 👍

**Comment by devlead** [comment]
@daveMueller your changes have been merged, thanks for your contribution 👍

**Pull request merged**